treeherder/perf/data_cycling/cyclers.py

```diff
@@ -0,0 +1,198 @@
+import logging
+from abc import ABC, abstractmethod
+from datetime import timedelta, datetime
+from typing import List
+
+from django.db import OperationalError, connection
+from django.db.backends.utils import CursorWrapper
+from django.db.models import Count
+
+from treeherder.config import settings
+from treeherder.model.models import Job, JobType, JobGroup, Machine
+from treeherder.perf.data_cycling.removal_strategies import RemovalStrategy
+from treeherder.perf.data_cycling.utils import has_valid_explicit_days
+from treeherder.perf.exceptions import NoDataCyclingAtAll, MaxRuntimeExceeded
+from treeherder.perf.models import PerformanceSignature, PerformanceAlertSummary
+from treeherder.services.taskcluster import TaskclusterModel, DEFAULT_ROOT_URL as root_url
+from .max_runtime import MaxRuntime
+from .signature_remover import PublicSignatureRemover
+
+logger = logging.getLogger(__name__)
+
+TREEHERDER = 'treeherder'
+PERFHERDER = 'perfherder'
+
+
+class DataCycler(ABC):
```

Comment:
I thought I had added a comment here yesterday but apparently I didn't. I think these commands need to stay in a `management/commands` directory in order to be registered. Also this is still cycling Treeherder data so not sure why its being moved into a perf folder.

Reply:
> I thought I had added a comment here yesterday but apparently I didn't. I think these commands need to stay in a `management/commands` directory in order to be registered.

I agree. Custom commands inheriting `BaseCommand` need to be placed in this dir. But I haven't moved the `cycle_data` command from it. I've only moved its dependencies, because the module became too big (500-600 hundred SLOC).

> Also this is still cycling Treeherder data so not sure why its being moved into a perf folder.

Good catch! Hm... I think I'll move the package inside `treeherder/model`. Would that do?

Reply:
Sure, that works.